packages/pds/src/api/com/atproto/account/create.ts

```diff
@@ -68,33 +63,44 @@ export default function (server: Server, ctx: AppContext) {
         }
       }
 
-      // Pre-register user before going out to PLC to get a real did
-      try {
-        await actorTxn.preregisterDid(handle, tempDid)
-      } catch (err) {
-        if (err instanceof UserAlreadyExistsError) {
-          throw new InvalidRequestError(`Handle already taken: ${handle}`)
-        }
-        throw err
+      // format create op, but don't send until we ensure the username & email are available
+      const plcCreate = await ctx.plcClient.formatCreateOp(
+        ctx.keypair,
+        recoveryKey || ctx.cfg.recoveryKey,
+        handle,
+        ctx.cfg.publicUrl,
+      )
```

Comment:
Kicking myself for not seeing this before 🤦 this was the answer all along!

Reply:
dude you say that, but I _wrote_ the damn thing & didn't think of it 🙃